fix: do not alter original telemetry context pt.1

stijnmoreels opened the pull request:
Fixes the first set of telemetry extensions w/ a telemetry context that does not alter the original passed-in context. Split in two parts as there are a lot of extensions/tests to be changed.

Relates to #545

fgheysels commented:
I was first wondering if we would have a test that verifies if the original context is not altered, and after reviewing some files, I indeed realized  :  "why should I even wonder" .  Thanks for adding the additional test! :)

stijnmoreels commented:
> I was first wondering if we would have a test that verifies if the original context is not altered, and after reviewing some files, I indeed realized : "why should I even wonder" . Thanks for adding the additional test! :)

Also simplified the telemetry context determination to a single location, so that it becomes easier to see where the context is 'copied from' or 'created anew'.
The test are mostly verifying these indeed, but I could have missed an overload in the process; hence why I simplified it, so that we know for sure that we're testing the same thing.
